routers/private/hook_pre_receive.go

```diff
@@ -439,6 +442,50 @@ func preReceivePullRequest(ctx *preReceiveContext, oldCommitID, newCommitID, ref
 	}
 }
 
+func preReceiveDeletePRHeadRef(ctx *preReceiveContext, refFullName string) {
+	if !ctx.opts.GitPushOptions.Bool("delete-pull-head-confirm", false) {
+		ctx.JSON(http.StatusForbidden, private.Response{
+			Err: "Warning: you are attempting to delete the head reference of a pull request: " + refFullName + ".\n" +
+				"PR head references can only be deleted for closed PRs and only by their author or the repository admin.\n" +
+				"If you are sure what you want to do this, push again with the push option `git push ... -o delete-pull-head-confirm=true` ",
+		})
+		return
+	}
+
+	pullIndexStr := strings.TrimPrefix(refFullName, git.PullPrefix)
+	pullIndexStr = strings.Split(pullIndexStr, "/")[0]
+	pullIndex, _ := strconv.ParseInt(pullIndexStr, 10, 64)
+	if pullIndex <= 0 {
+		return
+	}
+
+	pr, err := models.GetPullRequestByIndex(ctx.Repo.Repository.ID, pullIndex)
+	if err != nil {
+		log.Error("Unable to get PR #%d in %s Error: %v", pullIndex, ctx.Repo.Repository.FullName(), err)
+		ctx.JSON(http.StatusInternalServerError, private.Response{
+			Err: fmt.Sprintf("Unable to get PR #%d Error: %v", pullIndex, err),
+		})
+		return
+	}
+
+	if !pr.Issue.IsClosed {
+		ctx.JSON(http.StatusForbidden, private.Response{
+			Err: fmt.Sprintf("Error: Head references of open pull request: %s cannot be deleted.", refFullName),
+		})
+		return
+	}
+
+	if pr.IssueID == ctx.opts.UserID {
+		return
+	}
```

Comment:
I don't think this is needed unless I'm missing something? 🤔 
```suggestion
```

Reply:
for pull request author.

Reply:
1) `IssueID` is reference to the `issue` table not `user` so you can't really compare these values
2) I don't think PR creator should need this rights, imho only repo admin should be allowed to delete PR head as this is destructible to target repository